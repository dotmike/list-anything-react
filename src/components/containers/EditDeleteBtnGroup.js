import React from "react";
import PropTypes from "prop-types";
import { Button, Popup } from "semantic-ui-react";

const EditDeleteBtnGroup = ({ onEdit, onDelete }) => (
  <Button.Group icon size="medium">
    <Popup
      trigger={<Button icon="edit" color="blue" onClick={onEdit} />}
      content="Edit"
    />
    <Popup
      trigger={<Button icon="delete" color="red" onClick={onDelete} />}
      content="Delete"
    />
  </Button.Group>
);

EditDeleteBtnGroup.propTypes = {
  onEdit: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired
};

export default EditDeleteBtnGroup;
