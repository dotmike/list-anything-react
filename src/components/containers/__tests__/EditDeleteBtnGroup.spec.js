import React from "react";
import { shallow } from "enzyme";

import EditDeleteBtnGroup from "../EditDeleteBtnGroup";

const defaultProps = {
  showEdit: true,
  onEdit: () => {},
  showDelete: true,
  onDelete: () => {}
};

describe("EditDeleteBtnGroup", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<EditDeleteBtnGroup {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
