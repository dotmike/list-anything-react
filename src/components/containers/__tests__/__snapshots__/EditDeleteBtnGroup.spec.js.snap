// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditDeleteBtnGroup renders correctly 1`] = `
<ButtonGroup
  icon={true}
  size="medium"
>
  <Popup
    content="Edit"
    on="hover"
    position="top left"
    trigger={
      <Button
        as="button"
        color="blue"
        icon="edit"
        onClick={[Function]}
      />
    }
  />
  <Popup
    content="Delete"
    on="hover"
    position="top left"
    trigger={
      <Button
        as="button"
        color="red"
        icon="delete"
        onClick={[Function]}
      />
    }
  />
  <Confirm
    cancelButton="Cancel"
    confirmButton="OK"
    content="Are you sure?"
    onCancel={[Function]}
    onConfirm={[Function]}
    open={false}
  />
</ButtonGroup>
`;
